web/src/scripts/ipc/ipcActionEmitter.js

```diff
@@ -164,6 +178,24 @@ const ipcActionEmitter = (store) => {
     store.dispatch(setConsoleVisibility(!state.ui.consoleVisible))
   })
 
+  ipc.on(SHOULD_CLOSE_TAB, () => {
```

Comment:
Everything else looks good. My only request is to move this logic into an action. Now that it's being used in two places... probably should've done this earlier anyway.

I'm thinking you can put it in web/compositeFileActions.js ... the block of logic looks and behaves similarly to the openFile() function in there.

TabbedEditor.jsx and ipcActionEmitter.js can then dispatch the high-level action rather than duplicating functionality.